src/statusnotifieritem/statusnotifieritem.cpp

```diff
@@ -43,17 +43,22 @@ StatusNotifierItem::StatusNotifierItem(QString id, QObject *parent)
     mTitle("Test"),
     mStatus("Active"),
     mMenu(nullptr),
-    mMenuExporter(nullptr)
+    mMenuExporter(nullptr),
+    mSessionBus(QDBusConnection::connectToBus(QDBusConnection::SessionBus, mService))
 {
+    // we created a separate DBus connection, because multiple dbus services with _different_
+    // behavior are not supported on QDbus
```

Comment:
That was what I meant by the comment -- you can not have different objects registered for different services. Probably not worded the best way though.

Anyway the idea was taken from KDE's KNotification, so if Qt uses it too, that is even better.